### PR TITLE
Created main interfaces for bring

### PR DESCRIPTION
**Pull request by mykolad4**
Work under JavaDoc is in progress...

### CODE REVIEW

File: src/main/java/com/pertos/bring/context/ApplicationContext.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.pertos.bring.context;
+
+import com.pertos.bring.exception.NoSuchBeanException;
+import com.pertos.bring.exception.NoUniqueBeanException;
+
+import java.util.Map;
+
+public interface ApplicationContext {
+
+    <T> T getBean(Class<T> beanType) throws NoSuchBeanException, NoUniqueBeanException;
```
Comment (mitssev) [inline src/main/java/com/pertos/bring/context/ApplicationContext.java:10]: once exceptions defined here shouldn't they be checked ?

File: src/main/java/com/pertos/bring/reader/BeanDefinitionReader.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.pertos.bring.reader;
+
+import com.pertos.bring.exception.BeanDefinitionStoreException;
+
+public interface BeanDefinitionReader {
+
+    int loadBeanDefinitions(String location) throws BeanDefinitionStoreException;
```
Comment (mitssev) [inline src/main/java/com/pertos/bring/reader/BeanDefinitionReader.java:7]: int to return number of loaded BDs?

**Comment by mitssev** [comment]
@mykolad4 could you add some javadoc comments and annotations? Or will be done in the other task?

File: src/main/java/com/pertos/bring/context/ApplicationContext.java (hunk 1 above)
Comment (mykolad4) [reply src/main/java/com/pertos/bring/context/ApplicationContext.java:10]: I'm not sure it is a good idea to force our client to catch checked exceptions. Here is an example how Spring's ApplicationContext works:
`ApplicationContext context = SpringApplication.run(MockServerApplication.class, args);
 Object notExistedClass = context.getBean("notExistedClass");`
Application will be terminated with exception in logs:
`Exception in thread "main" org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'notExistedClass' available`

File: src/main/java/com/pertos/bring/reader/BeanDefinitionReader.java (hunk 2 above)
Comment (mykolad4) [reply src/main/java/com/pertos/bring/reader/BeanDefinitionReader.java:7]: Correct. Have no clue why we need it. But I stole it from Spring's implementation

File: src/main/java/com/pertos/bring/context/ApplicationContext.java (hunk 1 above)
Comment (mitssev) [reply src/main/java/com/pertos/bring/context/ApplicationContext.java:10]: than why are they defined in throws? ok.. nevermind

**Review by artem-dehnych — APPROVED** [review]

**Review by mitssev — APPROVED** [review]

**Review by Arrrmy21 — APPROVED** [review]

**Pull request merged**